### PR TITLE
[v8] Backport #11609

### PR DESCRIPTION
**Pull request by lxea**

### CODE REVIEW

**Review by xacrimon — APPROVED** [review]

File: lib/client/api_test.go [hunk 1]
```diff
@@ -500,3 +504,108 @@ func TestApplyProxySettings(t *testing.T) {
...
+	require.Equal(t, "VALID", client.getProxySSHPrincipal(), "ProxySSHPrincipal mismatch")
+
+	cfg.UseKeyPrincipals = false // causes xyz to be returned as key was not used
+
+	client, err = NewClient(cfg)
+	require.NoError(t, err)
+	require.Equal(t, "xyz", client.getProxySSHPrincipal(), "ProxySSHPrincipal mismatch")
+}
+
+func TestParseSearchKeywords(t *testing.T) {
```
Comment (codingllama) [inline lib/client/api_test.go:555]: This isn't really part of the backport, is it?

**Comment by xacrimon** [comment]
lgtm once @codingllama 's inquiries are taken care of

File: lib/client/api_test.go (hunk 1 above)
Comment (lxea) [reply lib/client/api_test.go]: Ah sorry, mistake with a merge

**Review by codingllama — APPROVED** [review]

File: lib/client/api_test.go (hunk 1 above)
Comment (codingllama) [reply lib/client/api_test.go]: No worries, same thing happened with me yesterday.

**Pull request merged**